base/reinterpretarray.jl

```diff
@@ -696,20 +696,24 @@ function iterate(cp::CyclePadding, state::Tuple)
 end
 
 """
-    Compute the location of padding in a type.
+    Compute the location of padding in an isbits datatype. Recursive over the fields of that type.
 """
-function padding(T)
-    padding = Padding[]
-    last_end::Int = 0
+@assume_effects :foldable function padding(T::DataType, baseoffset::UInt = UInt(0))
+    pads = Padding[]
+    last_end::Int = baseoffset
     for i = 1:fieldcount(T)
-        offset = fieldoffset(T, i)
+        offset = baseoffset + fieldoffset(T, i)
```

Comment:
```suggestion
        offset = baseoffset + Int(fieldoffset(T, i))
```
I am not really sure why fieldoffset is returning UInt, but probably because it not really used elsewhere, so I think it is a better API to stick with Int, since that is how Julia usually represents numbers.